src/storage/v3/expr.cpp

```diff
@@ -186,7 +186,7 @@ TypedValue ComputeExpression(DbAccessor &dba, const std::optional<memgraph::stor
   expr::SymbolGenerator symbol_generator(&symbol_table, identifiers);
   (std::any_cast<Expression *>(expr))->Accept(symbol_generator);
 
-  if (node_identifier.symbol_pos_ != -1) {
+  if (node_identifier.symbol_pos_ != -1 && v_acc.has_value()) {
```

Comment:
When can a vertex be nullopt? Based on the code I would say when we want to evaluate order by expresssions on edges. However I think that is wrong. When we are evaluating expressions on edges, we should always be possible to access the vertex we are using to iterate over its edges.

Reply:
It crashes without it. I'll add more tests and try to understand afterwards

Reply:
I have removed the check and made it an assert instead.
Crash was due to a nullopt vertex_accessor (which was possible within our workflow).
To summarize how we solved the root cause: we will always keep the vertex_accessor when sorting edges (that comes from a single node). An example where this is needed: an ordering based on a map on the node, where edges would be the key.

Reply:
You assert on the edge accessor, not on the vertex accessor.

Reply:
To explain for future us: two asserts were added, one on the `std::optional<VertexAccessor>` and one on the `std::optional<EdgeAccessor>`
As part of an another remark, the `VertexAccessor` no longer need to be optional so we are left with a single assert on `EdgeAccessor`